tests/test_cli.py

```diff
@@ -0,0 +1,220 @@
+    invalid_value = "wrong_one"
+    for f in flags:
+        base_one.extend([f, invalid_value])
+        p = Popen(base_one, stdout=PIPE, stderr=PIPE)
+        output, error = p.communicate()
+        assert p.returncode == 1
+        assert "unrecognized arguments:" in str(error)
+
+        base_one.pop()
+        p = Popen(base_one, stdout=PIPE, stderr=PIPE)
+        output, error = p.communicate()
+        assert p.returncode == 1
+        assert (
+            "Oops! File with queries was not found. Create it or set name properly."
+            in str(output)
+        )
+        base_one.pop()
+
+
+def test_cli_invalid_args_after_flags(cli_args: list) -> None:
+    """
+    Check behavior in case of invalid arguments after different flags
+    :param cli_args: base arguments
+    :return: None
+    """
+    invalid_args = cli_args
+    invalid_value = "wrong_one"
+
+    invalid_args.extend(["-q", invalid_value])
+    p = Popen(invalid_args, stdout=PIPE, stderr=PIPE)
+    output, error = p.communicate()
+    assert p.returncode == 1
+    assert (
+        "Oops! File with queries was not found. Create it or set name properly."
+        in str(output)
+    )
+    invalid_args[-1] = CLITestDefaultValues.CLI_QUERY_PATH
+
+    invalid_args.extend(["-v", invalid_value])
+    p = Popen(invalid_args, stdout=PIPE, stderr=PIPE)
+    output, error = p.communicate()
+    assert p.returncode == 1
+    assert "Vendors not found in queries file" in str(output)
+    invalid_args = invalid_args[:-2]
+
+    flags_extension = ["-nm", "-vs", "-sc", "-vs", "-ts"]
+    invalid_args.extend(flags_extension)
+
+    flags = ["-cm", "-sm", "-nw", "-vw", "-ht", "-tp", "-ml", "-tls"]
+    expected_invalid_errors = [
+        "argument -cm/--censys-max: invalid int value",
+        "argument -sm/--shodan-max: invalid int value",
+        "argument -nw/--nmap-workers: invalid int value",
+        "argument -vw/--vulners-workers: invalid int value",
+        "argument -ht/--host-timeout: invalid int value",
+        "argument -tp/--top-ports: invalid int value",
+        "argument -ml/--max-limit: invalid int value",
+    ]
+    for flag_arg, flag_output in zip(flags, expected_invalid_errors):
+        invalid_args.extend([flag_arg, invalid_value])
+        p = Popen(invalid_args, stdout=PIPE, stderr=PIPE)
+        output, error = p.communicate()
+        assert p.returncode == 1
+        assert flag_output in str(error)
+
+        invalid_args.pop()
+        p = Popen(invalid_args, stdout=PIPE, stderr=PIPE)
+        output, error_without_arg = p.communicate()
+        assert p.returncode == 1
+        assert "expected one argument" in str(error_without_arg)
+        invalid_args.pop()
+
+
+def test_cli_vendor_confidence(cli_args: list) -> None:
+    """
+    Check behavior in case of right vendor confidence
+    :param cli_args: base arguments
+    :return: None
+    """
+    base_args = cli_args
```

Comment:
I guess you think that by assigning a value of `cli_args` to `base_args` you will copy it into `base_args`, but that is not true.

```python3
#!/usr/bin/env python3

def test_cli_vendor_confidence(cli_args: list) -> None:
    base_args = cli_args
    base_args.extend(["this values was added in function"])

if __name__=="__main__":
    cli_args = [1, 2, 3]
    print(cli_args)
    test_cli_vendor_confidence(cli_args)
    print(cli_args)
```
Results:
```
[1, 2, 3]
[1, 2, 3, 'this values was added in function']
```
You just assign a link to `cli_args` into `base_args`. If you really want to create independent copy of `cli_args`, you need to use `copy` or forcely call to `list`.

Example:
```python3
def test_cli_vendor_confidence(cli_args: list) -> None:
    # Look
    base_args = list(cli_args)
    base_args.extend(["this values was added in function"])

if __name__=="__main__":
    cli_args = [1, 2, 3]
    print(cli_args)
    test_cli_vendor_confidence(cli_args)
    print(cli_args)
```
Results:
```
[1, 2, 3]
[1, 2, 3]
```
or 
```python3
#!/usr/bin/env python3
from copy import copy

def test_cli_vendor_confidence(cli_args: list) -> None:
    base_args = copy(cli_args)
    base_args.extend(["this values was added in function"])
```

Note: if you really want to modify cli_args, you can use it directly, without assigning. BUT, it is always better to check what will happen with you variable or data, so it's better to copy or create an independent element, as I said.

Reply:
SORRY! I just notice that this is a fixture! My bad! :)
The way you do it is okay in this case.